docs/components/button.md

```diff
@@ -121,6 +121,24 @@ KButton supports using an icon either before the text or without text.
 ```
 
 ## Theming
+KButton includes a utility class `.has-caret` which displays a dropdown caret to the right hand side. this is useful for buttons that control dropdowns and popovers. You can rotate the caret (active state) by applying `.is-active` as an additional class.
```

Comment:
Needing to know about a class is an implementation detail that we should obscure from users as you can't use the class outside k-button. I am thinking it would be easier to use a single prop `:isOpen` which would then toggle the `.has-caret` and `.is-active` classes. The absence of the prop would be "no caret".

Reply:
how about `hasCaret` prop? How would `is-open`/`is-active` be set then? Now you need two props instead of two classes. I thought about going the prop route but you still need a way to toggle the state

Reply:
I was thinking something like (pseudo code):
```js
let classes = ""
if (isOpen !== undefined) {
  classes = isOpen ? ".has-caret .is-active" : ".has-caret"
}
```

Reply:
Eh this seems like added complexity for a caret. I'd rather add the utility class to the general css (truncate, cursor, etc) instead